Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1

```diff
@@ -874,3 +874,194 @@ Describe 'Confirm-DiscoveryConfig Unit Tests'{
+                # Assert
+                $runResult | Should -Be $false
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Search-XMLChildNode -Scope It -Times 1 -Exactly
+            }
+            It 'Should succeed but not remove install.config settings if comparison files are different'{
+                # Arrange
+                $customObject = [PSCustomObject]@{
+                 Path = ".\"
+                 Owner = "TestOwner"
+                 Group = "TestUsers"
+                 AccessType = "Allow"
+                 Rights = "ModifyData"
+                 ObjectDifference = "<variable name = 'newitem' value = 'newvalue'/>"
+                }
+
+                Mock -ModuleName Install-Identity-Utilities -CommandName Search-XMLChildNode -MockWith {$true}
+
+                $nodesToSearch = @("tenants","replyUrls","claimsIssuerTenant","allowedTenants","registeredApplications","azureSecretName")
+                $xmlChildNodes = Get-XMLChildNodes -installConfigPath "install.config" -configSection "identity" -nodesToSearch $nodesToSearch -childNodeGetAttribute "name"
+
+                Mock -ModuleName Install-Identity-Utilities -CommandName Remove-XMLChildNodes -MockWith {}
+                Mock -ModuleName Install-Identity-Utilities -CommandName Add-XMLChildNodes -MockWith {}
+                Mock -ModuleName Install-Identity-Utilities -CommandName Compare-Object -MockWith {$customObject}
+
+                # Act
+                $runResult = Migrate-AADSettings -installConfigPath "install.config" -azureConfigPath "azuresettings.config" -nodesToSearch $nodesToSearch
+                
+                # Assert
+                $runResult | Should -Be $true
+                $xmlChildNodes.Count | Should -Be 6
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Search-XMLChildNode -Scope It -Times 1 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Remove-XMLChildNodes -Scope It -Times 2 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Add-XMLChildNodes -Scope It -Times 1 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Compare-Object -Scope It -Times 1 -Exactly
+            }
+        }
+    }
+}
+Describe 'Migrate-AADSettings' -Tag 'Integration'{
+    Context 'Migrating AAD Settings using Integration Tests'{
+        InModuleScope Install-Identity-Utilities{
+            It 'Should successfully run the migration'{
+                 # Arrange 
+                 # Get the contents of the files for testing from the tests folder
+                 # Add them to the powershell TestDrive which cleans up after each test leaving the tests folder clean
+                 $testInstallFile = 'testInstall.config'
+                 $testAzureFile = 'testAzure.config'
+                 $installConfigPath = "$($TestDrive)\$($testInstallFile)"
```

Comment:
What is TestDrive? is that something built into pester?

Reply:
built into powershell. https://github.com/pester/Pester/wiki/TestDrive